addons/account/wizard/account_merge_wizard.py

```diff
@@ -0,0 +1,319 @@
+        wiz = self.env['base.partner.merge.automatic.wizard'].new()
+        wiz._update_foreign_keys_generic('account.account', accounts_to_remove, account_to_merge_into)
+
+        # 3.2: Update Reference and Many2OneReference fields that reference account.account
+        wiz._update_reference_fields_generic('account.account', accounts_to_remove, account_to_merge_into)
+
+        # Step 4: Remove merged accounts
+        self.env.invalidate_all()
+        self.env.cr.execute(SQL(
+            """
+             DELETE FROM account_account
+              WHERE id IN %(account_ids_to_delete)s
+            """,
+            account_ids_to_delete=tuple(accounts_to_remove.ids),
+        ))
+
+        # Clear ir.model.data ormcache
+        self.env.registry.clear_cache()
+
+        # Step 5: Write company_ids and codes on the account
+        for company, code in code_by_company.items():
+            account_to_merge_into.with_company(company).sudo().code = code
+
+        account_to_merge_into.sudo().company_ids = company_ids_to_write
+
+
+class AccountMergeWizardLine(models.TransientModel):
+    _name = 'account.merge.wizard.line'
+    _description = "Account merge wizard line"
+    _order = 'sequence, id'
+
+    wizard_id = fields.Many2one(
+        comodel_name='account.merge.wizard',
+        required=True,
+        ondelete='cascade',
+    )
+    sequence = fields.Integer()
+    grouping_key = fields.Char()
+    display_type = fields.Selection(
+        selection=[
+            ('line_section', "Section"),
+            ('account', "Account"),
+        ],
+        required=True,
+    )
+    is_selected = fields.Boolean()
+    account_id = fields.Many2one(
+        string="Account",
+        comodel_name='account.account',
+        ondelete='cascade',
+        # We do this rather than readonly=True, so that the user can edit the account from the Many2one widget,
+        # without being given a dropdown with other accounts to choose from.
+        domain="[('id', '=', False)]",
+    )
+    company_ids = fields.Many2many(
+        string="Companies",
+        related='account_id.company_ids',
+    )
+    info = fields.Char(
+        string='Info',
+        compute='_compute_info',
+        store=True,
+        help="Contains either the section name or error message, depending on the line type."
+    )
+    account_has_hashed_entries = fields.Boolean(compute='_compute_account_has_hashed_entries', store=True)
+
+    @api.depends('account_id')
+    def _compute_account_has_hashed_entries(self):
+        """ This is an optimization to avoid having to re-check which accounts have hashed entries. """
+        query = self.env['account.move.line'].sudo()._search([('account_id', 'in', self.account_id.ids), ('move_id.inalterable_hash', '!=', False)])
+        query_result = self.env.execute_query(query.select(SQL('DISTINCT account_move_line.account_id')))
+        accounts_with_hashed_entries_ids = [r[0] for r in query_result]
```

Comment:
Make the line below O(n) vs O(n²)
```suggestion
        accounts_with_hashed_entries_ids = {r[0] for r in query_result}
```